src/task/ReceiveVideoInputTask.ts

```diff
@@ -24,40 +23,35 @@ export default class ReceiveVideoInputTask extends BaseTask {
     height: number,
     frameRate: number
   ): Promise<void> {
-    const trackSettings = mediaStreamTrack.getSettings();
-    let videoQualitySettings: VideoQualitySettings;
+    const videoQualitySettings = isContentAttendee
+      ? this.context.meetingSessionConfiguration.meetingFeatures.contentMaxResolution
+      : this.context.meetingSessionConfiguration.meetingFeatures.videoMaxResolution;
 
-    if (isContentAttendee) {
-      videoQualitySettings = this.context.meetingSessionConfiguration.meetingFeatures
-        .contentMaxResolution;
-    } else {
-      videoQualitySettings = this.context.meetingSessionConfiguration.meetingFeatures
-        .videoMaxResolution;
-    }
     if (
-      width > videoQualitySettings.videoWidth ||
-      height > videoQualitySettings.videoHeight ||
-      frameRate > videoQualitySettings.videoFrameRate
+      !isContentAttendee &&
+      width <= videoQualitySettings.videoWidth &&
+      height <= videoQualitySettings.videoHeight &&
+      frameRate <= videoQualitySettings.videoFrameRate
     ) {
-      const constraint: MediaTrackConstraints = {
-        width: { ideal: videoQualitySettings.videoWidth },
-        height: { ideal: videoQualitySettings.videoHeight },
-        frameRate: { ideal: videoQualitySettings.videoFrameRate },
-      };
-      this.context.logger.warn(
-        `Video track (content = ${isContentAttendee}) will be constrained to: ${JSON.stringify(
-          constraint
-        )} to remain below configured video quality settings, trackSettings: ${JSON.stringify(
-          trackSettings
-        )}`
+      // Skip applying constraints if the current video track settings are already
+      // lower than the limit except for content share. Always apply constraint for
+      // content attendees because content share resolution may change.
+      return;
+    }
+
+    const constraint: MediaTrackConstraints = {
+      resizeMode: { ideal: 'none' },
```

Comment:
This is interesting, whats the effect of not setting this? The description on MDN is:
```
none means that the user agent uses the resolution provided by the camera, its driver or the OS
```
But seems like it will still scale. Maybe a comment could help clarify other similar confusion in the future.

Reply:
Chrome would scale content share to the maximum resolution within max width and height without this configured. This is not what we want. Adding this resize mode with ideal will make browser not doing scaling if resolution is within limits, otherwise it will scale to meet max requirements. Firefox and safari don't have this option, but the two browsers don't have the unexpected scaling behavior and is not influenced by this.

Reply:
Gotcha, perfect information.